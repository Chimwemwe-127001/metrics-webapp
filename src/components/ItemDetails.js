import React from 'react';
import { useParams } from 'react-router';
import { useSelector } from 'react-redux';

const ItemDetails = () => {
  const countriesList = useSelector((state) => state.covidReducer.oneCountry);
  const { name } = useParams();
  const countries = countriesList ? countriesList.find((country) => country.name === name) : 'Still fetching';
  const {
    confirmedCases,
    recovered,
    openCases,
    allDeaths,
    id,
  } = countries;
  return (
    <div className="detail-info" key={id}>
      <h1>
        {name}
        <span>
          {confirmedCases + openCases}
          {' '}
          cases
        </span>
      </h1>
      <ul className="ul">
        <li className="li d-flex">
          <p>Total cases confirmed</p>
          <p>
            {confirmedCases}
          </p>
        </li>
        <li className="li d-flex">
          <p>Total Deaths confirmed</p>
          <p>
            {allDeaths}
          </p>
        </li>
        <li className="li d-flex">
          <p>Total cases open</p>
          <p>
            {openCases}
          </p>
        </li>
        <li className="li d-flex">
          <p>Total recovered</p>
          <p>
            {recovered}
          </p>
        </li>
      </ul>
    </div>
  );
};

export default ItemDetails;
