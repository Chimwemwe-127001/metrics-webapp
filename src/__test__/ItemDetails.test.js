import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ItemDetails from '../components/ItemDetails';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('testing the app', () => {
  it('testing ItemDetails UI', () => {
    act(() => {
      render(
        <Provider store={store}>
          <ItemDetails />
        </Provider>, container,
      );
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
