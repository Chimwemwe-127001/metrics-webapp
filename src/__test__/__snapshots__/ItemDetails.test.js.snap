// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the app testing ItemDetails UI 1`] = `
"<div class=\\"detail-info\\">
  <h1><span>NaN cases</span></h1>
  <ul class=\\"ul\\">
    <li class=\\"li d-flex\\">
      <p>Total cases confirmed</p>
      <p></p>
    </li>
    <li class=\\"li d-flex\\">
      <p>Total Deaths confirmed</p>
      <p></p>
    </li>
    <li class=\\"li d-flex\\">
      <p>Total cases open</p>
      <p></p>
    </li>
    <li class=\\"li d-flex\\">
      <p>Total recovered</p>
      <p></p>
    </li>
  </ul>
</div>"
`;
